nixos/modules/hardware/i2c.nix

```diff
@@ -0,0 +1,42 @@
+{ config, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.hardware.i2c;
+in
+
+{
+  options.hardware.i2c = {
+    enable = mkEnableOption ''
+      i2c devices support. By default access is granted to users
+      in the "i2c" group (will be created if non-existent)
+    '';
+
+    group = mkOption {
+      type = types.str;
+      default = "i2c";
+      description = ''
+        Grant access to i2c devices (/dev/i2c-*) to users in this group.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    boot.kernelModules = [ "i2c-dev" ];
+
+    users.groups = mkIf (cfg.group == "i2c") {
+      i2c = { };
+    };
+
+    services.udev.extraRules = ''
+      # allow group ${cfg.group} use of i2c devices
+      ACTION=="add", KERNEL=="i2c-[0-9]*", GROUP="${cfg.group}", MODE="666"
```

Comment:
Mode 666 will give _all_ users access. Don't we want 660 or something?

Reply:
Obvious mistake, thank you for spotting it.